### PR TITLE
Replace case.mat with grid.pkl

### PR DESCRIPTION
**Pull request by jenhagg**
[Pull Request doc](https://breakthrough-energy.github.io/docs/user/git_guide.html#d-pull-request)

### Purpose
Pass a grid.pkl file as input to REISE.jl instead of case.mat. Partially addresses #668 along with https://github.com/Breakthrough-Energy/REISE.jl/pull/185

### What the code is doing
Remove `export_case_mat` which is now in REISE.jl temporarily. Preserve index names in transform_grid.py so operations on it after unpickling work as expected.

### Testing
Ran a scenario in a container.

### Time estimate
10 min

### CODE REVIEW

**Review by rouille — APPROVED** [review]
Looks good

File: powersimdata/scenario/execute.py [hunk 1]
```diff
@@ -204,20 +203,12 @@ def __init__(self, data_access, scenario_info, grid, ct):
...
-        with self._data_access.write(file_path, save_local=False) as f:
-            savemat(f, mpc, appendmat=False)
-
-        if mpc_storage is not None:
-            with self._data_access.write(storage_file_path, save_local=False) as f:
-                savemat(f, mpc_storage, appendmat=False)
+    def prepare_grid(self):
+        """Prepare grid for simulation."""
+        print("--> Preparing grid data")
+        dest_path = "/".join([self.REL_TMP_DIR, "grid.pkl"])
```
Comment (BainanXia) [inline powersimdata/scenario/execute.py:209]: Not sure whether we should use `os.join` here or it is taken care of in `_data_access` already.

File: powersimdata/scenario/execute.py [hunk 2]
```diff
@@ -258,11 +246,8 @@ def prepare_demand_flexibility_parameters_file(self):
...
             else 0
         ]
-
-        # Convert the dictionary to a DataFrame to more easily save as a .csv file
         params = pd.DataFrame.from_dict(params)
 
-        # Save the parameters as a .csv file and move the file to the correct location
-        file_path = "/".join([self.REL_TMP_DIR, "demand_flexibility_parameters.csv"])
-        with self._data_access.write(file_path, save_local=False) as f:
+        dest_path = "/".join([self.REL_TMP_DIR, "demand_flexibility_parameters.csv"])
```
Comment (lanesmith) [inline powersimdata/scenario/execute.py:251]: Similar comment to @BainanXia's above.

**Review by lanesmith — APPROVED** [review]
Looks good to me. Thanks!

File: powersimdata/scenario/execute.py (hunk 1 above)
Comment (rouille) [reply powersimdata/scenario/execute.py:209]: The same syntax is used in `prepare_profile`. I believe it is taken care of in `DataAccess`. @jenhagg will know for sure.

Comment (jenhagg) [reply powersimdata/scenario/execute.py:209]: Right - we use `"/".join` here since we're dealing with a pyfilesystem path that is os independent.

**Review by BainanXia — APPROVED** [review]

**Pull request merged**